File: Leetcode_2460.java
import java.util.*;

class Solution {
    public int[] applyOperations(int[] nums) {
        // Step 1: Apply doubling rule
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] == nums[i + 1]) {
                nums[i] *= 2;
                nums[i + 1] = 0;
            }
        }

        // Step 2: Move all 0s to the end (two-pointer approach)
        int[] result = new int[nums.length];
        int index = 0;

        for (int num : nums) {
            if (num != 0) {
                result[index++] = num;
            }
        }

        return result;
    }
}

File: Leetcode_2460.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        # Step 1: Apply the doubling rule
        for i in range(len(nums) - 1):
            if nums[i] == nums[i + 1]:
                nums[i] = nums[i] * 2
                nums[i + 1] = 0

        # Step 2: Move all zeros to the end
        nums = [x for x in nums if x != 0] + [0] * nums.count(0)
        return nums
